application-analytics-ui/src/main/resources/Analytics/Code/Macros/VelocityMacros.xml

```diff
@@ -61,5 +61,104 @@
   #end
   #set ($date = $startDate + ',' + $endDate)
 #end
+
+#macro(setParameterOnAction $parameter)
+&lt;div class="analyticsActions"&gt;
+  &lt;div class="analyticsRowEvolution"  data-parameter=$escapetool.url($parameter)&gt;
+    &lt;a class="action" title="$escapetool.xml($services.localization.render(
+      'analytics.mostViwedMacros.action.rowEvolution'))" href="#"&gt;
+      &lt;span class="action-icon"&gt;$services.icon.renderHTML('eye')&lt;/span&gt;&lt;span class="action-label"&gt;
+      $escapetool.xml($services.localization.render('analytics.action.label.rowEvolution'))&lt;/span&gt;
+    &lt;/a&gt;
+  &lt;/div&gt;
+&lt;/div&gt;
+#end
+
+#macro(durationSelect $className $options)
+&lt;select class="$className form-control"&gt;
+  #foreach($option in $options)
+    &lt;option value="$option"&gt;$option&lt;/option&gt;
+  #end
+&lt;/select&gt;
+#end
+
+
+#macro(durationOption $value)
+  &lt;option value="$value"&gt;
+    $services.localization.render('analytics.rowEvolution.modal.' + $value.toLowerCase() + '.label')
+  &lt;/option&gt;
+#end
+
+#macro(durationSelectList $class $options)
+  &lt;select class="$class form-control"&gt;
+    #foreach($option in $options)
+      &lt;option value="$option"&gt;$option&lt;/option&gt;
+    #end
+  &lt;/select&gt;
+#end
+
+#macro(fieldsSelect $fields)
+  &lt;select class="form-control analyticsApplicationRowEvolutionField"&gt;
```

Comment:
Same, you don't need to repeat the `analyticsApplication` suffix. Maybe `rowEvolutionField`?